pkgs/by-name/dy/dynamodb-local/package.nix

```diff
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchurl
+, jre
+, makeBinaryWrapper
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dynamodb-local";
+  version = "2023-12-14";
+
+  src = fetchurl {
+    url = "https://d1ni2b6xgvw0s0.cloudfront.net/v2.x/dynamodb_local_${finalAttrs.version}.tar.gz";
+    hash = "sha256-F9xTcLNAVFVbH7l0FlMuVNoLBrJS/UcHKXTkJh1n40w=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ makeBinaryWrapper ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/share/${finalAttrs.pname}
+    cp -r DynamoDBLocal* $out/share/${finalAttrs.pname}
+
+    makeBinaryWrapper ${jre}/bin/java $out/bin/${finalAttrs.pname} \
+      --add-flags "-jar $out/share/${finalAttrs.pname}/DynamoDBLocal.jar"
```

Comment:
Please don't rely on `pname`, this creates a binding that goes too far. It's better to use a string literal.

Reply:
also in mainProgram write the literal?

Reply:
Yes please